platform-services-handler/test/test.js

```diff
@@ -15,12 +15,344 @@
+    logMessage = "error finding service ";
+    var responseObject = {
+      statusCode : 400,
+      body:{message : "error finding service"}
+    };
+    AWS.mock("SQS","sendMessageBatch", (params, cb)=>{
+      return cb(null, callbackObj)
+    });
+    logStub = sinon.stub(logger, 'error', spy);
+    reqStub = sinon.stub(request, "Request", (obj)=>{
+      return obj.callback(null, responseObject, null);
+    });
+    var callFunction = index.handler(event, context, callback);
+    var logCheck = logStub.args[0][0].includes(logMessage);
+    logStub.restore();
+    reqStub.restore();
+    AWS.restore("SQS");
+    assert.isTrue(logCheck);
+  });
+  
+  it("should indicate error if response status code is 200 with body.data undefined and kinesis data is defined for complted endingEvent", ()=>{
+    event.Records[0].kinesis.data = "ew0KCSJJdGVtIjogew0KCQkiRVZFTlRfSUQiOiB7DQoJCQkiUyI6ICI0NWZlOGM4Mi1mZjFkLWIzMWQtMmFhNS1hMjJkMDkxMWI3ZWMiDQoJCX0sDQoJCSJUSU1FU1RBTVAiOiB7DQoJCQkiUyI6ICIyMDE3LTA2LTI2VDE3OjU0OjI2OjA4NiINCgkJfSwNCgkJIlNFUlZJQ0VfQ09OVEVYVCI6IHsNCgkJCSJTIjogIntcInNlcnZpY2VfdHlwZVwiOlwiYXBpXCIsXCJydW50aW1lXCI6XCJub2RlanNcIixcImFkbWluX2dyb3VwXCI6XCJuYW1lPWQmbmFtZT1iJm5hbWU9YSZuYW1lPWImbmFtZT11JlwifSINCgkJfSwNCgkJIkVWRU5UX0hBTkRMRVIiOiB7DQoJCQkiUyI6ICJKRU5LSU5TIg0KCQl9LA0KCQkiRVZFTlRfTkFNRSI6IHsNCgkJCSJTIjogIk9OQk9BUkRJTkciDQoJCX0sDQoJCSJTRVJWSUNFX05BTUUiOiB7DQoJCQkiUyI6ICJ0ZXN0OCINCgkJfSwNCgkJIkVWRU5UX1NUQVRVUyI6IHsNCgkJCSJTIjogIkNPTVBMRVRFRCINCgkJfSwNCgkJIkVWRU5UX1RZUEUiOiB7DQoJCQkiUyI6ICJTRVJWSUNFX0NSRUFUSU9OIg0KCQl9LA0KCQkiVVNFUk5BTUUiOiB7DQoJCQkiUyI6ICJzdmNfY3B0X2pua19hdXRoX3ByZCINCgkJfSwNCgkJIkVWRU5UX1RJTUVTVEFNUCI6IHsNCgkJCSJTIjogIjIwMTctMDUtMDVUMDY6MDY6Mzc6NTMzIg0KCQl9LA0KCQkiQUFBIjogew0KCQkJIk5VTEwiOiB0cnVlDQoJCX0sDQoJCSJCQkIiOiB7DQoJCQkiUyI6ICJ2YWwiDQoJCX0NCgl9LA0KCSJSZXR1cm5Db25zdW1lZENhcGFjaXR5IjogIlRPVEFMIiwNCgkiVGFibGVOYW1lIjogIkV2ZW50c19EZXYiDQp9"
+    logMessage = 'error finding service ';
+    var responseObject = {
+      statusCode : 200
+    };
+    var invalidArray = ["", null, undefined];
+    var bodyObj="{\"data\":[]}";
+    AWS.mock("SQS","sendMessageBatch", (params, cb)=>{
+      return cb(null, callbackObj)
+    });
+    logStub = sinon.stub(logger, 'error', spy);
+    for(i in invalidArray){
+      reqStub = sinon.stub(request, "Request", (obj) => {
+        bodyObj.data = invalidArray[i];
+        var body = JSON.stringify(bodyObj);
+        return obj.callback(null, responseObject, bodyObj);
+      });
+      var callFunction = index.handler(event, context, callback);
+      var logCheck = logStub.args[0][0].includes(logMessage);
+      reqStub.restore();
+    };
+    AWS.restore("SQS");
+    logStub.restore();
+    assert.isTrue(logCheck);
+  });
+
+  it("should indicate sucess from 200 response for the kinesis data with completed endingEvent",()=>{
```

Comment:
Kinesis input validation cases are covered. We need to also add test cases specific to the handler functionalities. Like how different Events are handled, what is the back-end service being triggered, How failure events are queued etc?

Reply:
You may need to refactor the index.js file to add unit test cases to cover all scenarios.

Reply:
refactoring of index.js is not included in this story

Reply:
We need to get to 90% code coverage, so if that requires code refactoring, we will need to do that.